packages/current-branch-name/src/__tests__/currentBranchName.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { currentBranchName } from "../currentBranchName";
+import { TestGitHubActionContext } from "@octopusdeploy/shared-action-utils";
+
+test("run with default main branch outputs correctly", () => {
+    const context = new TestGitHubActionContext();
+    context.addInput("github_ref", "refs/heads/main");
+    context.addInput("github_head_ref", "");
+    context.addInput("github_event_name", "workflow_dispatch");
+    context.addInput("replacement_strings", "/");
+    context.addInput("replacement_value", "-");
+    currentBranchName(context);
+
+    const outputs = context.getOutputs();
+    expect(outputs).toEqual({
+        branch_name: "main",
+    });
+});
+
+test("run with non default dev branch outputs correctly", () => {
+    const context = new TestGitHubActionContext();
+    context.addInput("github_ref", "refs/heads/dev");
+    context.addInput("github_head_ref", "");
+    context.addInput("github_event_name", "workflow_dispatch");
+    context.addInput("replacement_strings", "/");
+    context.addInput("replacement_value", "-");
+    currentBranchName(context);
+
+    const outputs = context.getOutputs();
+    expect(outputs).toEqual({
+        branch_name: "dev",
+    });
+});
+
+test("run with tag inputs outputs correctly", () => {
+    const context = new TestGitHubActionContext();
+    context.addInput("github_ref", "refs/tags/v1");
+    context.addInput("github_head_ref", "");
+    context.addInput("github_event_name", "workflow_dispatch");
+    context.addInput("replacement_strings", "/");
+    context.addInput("replacement_value", "-");
+    currentBranchName(context);
+
+    const outputs = context.getOutputs();
+    expect(outputs).toEqual({
+        branch_name: "v1",
+    });
+});
+
+test("run with pull request inputs outputs correctly", () => {
+    const context = new TestGitHubActionContext();
+    context.addInput("github_ref", "refs/pull/2/merge");
+    context.addInput("github_head_ref", "test;env;#");
```

Comment:
Would we normally expect `github_head_ref` to have a prefix on it e.g. `refs/heads/`?

Reply:
Correct, this was verified by outputting the inputs in a GitHub action run